### PR TITLE
Issue #39: Fixed a date comparison error, attempts to fix issues in the membership model.

### PR DESCRIPTION
**Pull request by lilliputten**

### CODE REVIEW

File: dds_registration/models.py [hunk 1]
```diff
@@ -122,7 +138,7 @@ class Event(models.Model):
     @property
     def can_register(self):
         today = date.today()
-        return self.registration_open >= today and (not self.registration_close or today <= self.registration_close)
```
Comment (lilliputten) [inline dds_registration/models.py:125]: It was my careless mistake.

File: dds_registration/models.py [hunk 2]
```diff
@@ -109,7 +124,8 @@ class Event(models.Model):
     description = models.TextField(blank=True)
     public = models.BooleanField(default=False)
     registration_open = models.DateField(auto_now_add=True, help_text='Date registration opens (inclusive)')
-    registration_close = models.DateField(blank=True, help_text='Date registration closes (inclusive)')
+    # Returned `null=True` here to allow to store an 'empty' dates. Is that ok?
```
Comment (cmutel) [inline dds_registration/models.py]: No, every event needs to have a close of registration - at the latest when the event finishes.

File: dds_registration/models.py (hunk 1 above)
Comment (cmutel) [reply dds_registration/models.py:125]: See above, we can assume that `registration_close` exists

File: dds_registration/models.py [hunk 3]
```diff
@@ -86,14 +86,29 @@ def __init__(self, *args, **kwargs):
 
 
 class Membership(models.Model):
-    user = models.ForeignKey(User, related_name='registrations', on_delete=models.CASCADE)
-    started = models.IntegerField(default=lambda: date.today().year)
-    until = models.IntegerField(default=lambda: date.today().year)
+    user = models.ForeignKey(User, related_name='memberships', on_delete=models.CASCADE)
```
Comment (cmutel) [inline dds_registration/models.py:91]: Please revert back to previous behaviour, it was chosen like that for a reason.

File: dds_registration/models.py (hunk 2 above)
Comment (lilliputten) [reply dds_registration/models.py]: > No, every event needs to have a close of registration - at the latest when the event finishes.

But in this case we should remove `blank=True` too, right?

File: dds_registration/models.py (hunk 3 above)
Comment (lilliputten) [reply dds_registration/models.py:91]: ~~Can you pls fix your code and provide its working version then? Probably, it'd be better to drop my fixes and to use your own updated revision.~~ Tried to make these changes by myself. See below.

**Comment by lilliputten** [comment]
Tried to make fixes by myself again. See it here: https://github.com/Depart-de-Sentier/dds_registration/pull/59/commits/e082f88121583948b5540a6d27560a9ea389d94c. Check it pls.

File: dds_registration/models.py (hunk 2 above)
Comment (cmutel) [reply dds_registration/models.py]: Yes, thanks.

File: dds_registration/models.py [hunk 4]
```diff
@@ -86,14 +88,22 @@ def __init__(self, *args, **kwargs):
...
+    # NOTE: Using different date comparsion method, using real dates, not only years...
+    started = models.IntegerField(default=this_year)
+    #  # OLD CODE: It causes an error during migration creation: `ValueError: Cannot serialize function: lambda`
+    #  started = models.IntegerField(default=lambda: date.today().year)
+    #  until = models.IntegerField(default=lambda: date.today().year)
     honorary = models.BooleanField(default=False)
 
+    @property
+    def until(self):
```
Comment (cmutel) [inline dds_registration/models.py]: Please revert this, one could have membership from e.g. 2020 to 2022 but they aren't active now. `until` needs to be a column.

File: dds_registration/models.py [hunk 5]
```diff
@@ -86,14 +88,22 @@ def __init__(self, *args, **kwargs):
...
+        # XXX: Should `until` always be the next year?
+        return self.started + 1 if self.started else None
+
     @property
     def active(self) -> bool:
-        return date.today().year <= self.until
+        return this_year() <= self.until
```
Comment (cmutel) [inline dds_registration/models.py:98]: This change is unnecessary but harmless...

**Comment by cmutel** [comment]
@lilliputten Please merge when you feel the questions raised have been answered. See also my commit.

**Pull request merged**